plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager, ServerTokenManager } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  AuthorizeRequestOptions,
+  DiscoveryApi,
+  PermissionClient,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * A server side {@link @backstage/plugin-permission-common#PermissionClient}
+ * that allows all backend-to-backend requests.
+ * @public
+ */
+export class ServerPermissionClient extends PermissionClient {
+  private readonly serverTokenManager: TokenManager;
+
+  constructor(options: {
+    discoveryApi: DiscoveryApi;
+    configApi: Config;
+    serverTokenManager: TokenManager;
+  }) {
+    const { discoveryApi, configApi, serverTokenManager } = options;
+    super({ discoveryApi, configApi });
+
+    if (this.enabled && !(serverTokenManager instanceof ServerTokenManager)) {
```

Comment:
Sup with this check? :grin: Doesn't quite feel like it's done in the right place, and we should totally allow other taken managers than just our built-in server one. In general I find that `instanceof` is really tricky and I find it best to avoid it.

I see the trickyness though, and can't really find a clean solution that doesn't add otherwise unnecessary methods. Perhaps we can at least flip it around for now and check `tokenManager instanceof ServerTokenManager.noop().constructor`?

Reply:
Changed to your suggestion and slapped a `TODO` on it as you mentioned in your comment above.